### ISSUE TITLE
M1 Macbook. Apple Silicon Support

### ISSUE THREAD

**Issue by qisaw** [aliyun/terraform-provider-alicloud#3660]
Apple Silicon is not supported by this provider.

### Terraform Version
```
Terraform v1.0.0
on darwin_arm64
```

### Affected Resource(s)
- all

### Panic Output
```
| 
│ Error: Incompatible provider version
│
│ Provider registry.terraform.io/aliyun/alicloud v1.124.3 does not have a
│ package available for your current platform, darwin_arm64.
│
│ Provider releases are separate from Terraform CLI releases, so not all
│ providers are available for all platforms. Other versions of this provider
│ may have different platforms supported.
|
```

### Expected Behavior
Terraform should initalize correctly on apple silicon macbooks

### Actual Behavior
An Error is thrown

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform init` on any apple silicon macbook

**Comment by goooogs** [comment]
Your macbook's platform is not `darwin_arm64`, it's `darwin_amd64`.
Download the correct terraform version from https://www.terraform.io/downloads.html.

**Comment by qisaw** [comment]
No, new Apple silicon MacBooks use Apple’s ARM based chips. The platform is definitely darwin_arm64. 

https://www.macrumors.com/guide/apple-silicon/

You can run the amd version of terraform currently with Rosetta 2 though. But would be nice to be able to run it natively on my machine.

**Comment by goooogs** [comment]
> No, new Apple silicon MacBooks use Apple’s ARM based chips. The platform is definitely darwin_arm64.
> 
> https://www.macrumors.com/guide/apple-silicon/
> 
> You can run the amd version of terraform currently with Rosetta 2 though. But would be nice to be able to run it natively on my machine.

My MacBook Chip is Appple M1.
The terraform was installed by `brew install terraform`, it's `darwin_arm64`.
I use aliyun/alicloud provider, and got this error.

I noticed that aliyun/alicloud provider and terraform are `darwin_amd64` on my another Intel chip MacBook, it works fine. But in M1 chip MacBook, terraform is `darwin_arm64`, it can not find aliyun/alicloud provider with `darwin_arm64`.

After I manually download terraform `darwin_amd64` from https://www.terraform.io/downloads.html (find aliyun/alicloud provider with `darwin_amd64),` all works fine for me.

This is a temporary solution.

Overall, you are right, it seems Apple Silicon is not supported by this provider now.
